x-pack/plugins/infra/public/components/log_stream/index.tsx

```diff
@@ -100,10 +105,26 @@ Read more at https://github.com/elastic/kibana/blob/master/src/plugins/kibana_re
   const parsedHeight = typeof height === 'number' ? `${height}px` : height;
 
   // Component lifetime
-  useMount(() => {
-    loadSourceConfiguration();
-    fetchEntries();
-  });
+  useDeepCompareEffect(() => {
+    const shouldReloadSourceConfiguration =
+      isUndefined(prevProps) || sourceId !== prevProps.sourceId;
+
+    const shouldReloadEntries =
+      isUndefined(prevProps) ||
+      shouldReloadSourceConfiguration ||
+      prevProps.startTimestamp !== startTimestamp ||
+      prevProps.endTimestamp !== endTimestamp ||
+      prevProps.query !== query ||
+      prevProps.center !== center;
+
+    if (shouldReloadSourceConfiguration) {
+      loadSourceConfiguration();
+    }
+
+    if (shouldReloadEntries) {
+      fetchEntries();
+    }
+  }, [prevProps, props, loadSourceConfiguration, fetchEntries]);
```

Comment:
Because of the way the defaults are set later during destructing but by using the original `props` here this wouldn't use the correct default for properties like `sourceId`. Wouldn't the below be just as correct since the functions already close over the relevant props and change their identity when needed?

```suggestion
  useEffect(() => {
    loadSourceConfiguration();
  }, [loadSourceConfiguration]);

  useEffect(() => {
    fetchEntries();
  }, [fetchEntries]);

```

Reply:
> Wouldn't the below be just as correct since the functions already close over the relevant props and change their identity when needed?

Yeah that should do the trick 👍